docs/development/decisions/0007-shared-sessions.adoc

```diff
@@ -0,0 +1,74 @@
+= 6. Shared user sessions
+
+Date: 2024-11-13
+
+== Status
+
+Draft
+
+== Context
+
+We need shared sessions between containers. In dev we only have one instance of the organisation-app that handles user
+session. In production we'll be scaling app the organisation-app depending on current demand, so it will never be a
+single instance.
+
+We need sessions to be stored outside of the running container (current behaviour), to provide users with a seamless
+experience.
+
+There are no Non Functional Requirements provided by the stakeholders, so it's difficult to estimate the number of
+concurrent users we have to to prepare for. We've ballpark estimated it to be anything between 50k to 500k, likely
+double that at the initial rollout. Customer expects 3.5m active users within first 2 years.
+
+=== Option 1: Sticky sessions
+
+Sticky sessions can be enabled at the infrastructure level, so user's requests are handled by a single container.
+
+Advantages:
+
+* Easy to set up
+* No code changes required
+* No additional cost
+
+Disadvantages:
+
+* User journeys can be disrupted by sub-optimal distribution across containers, especially during container restarts, i.e. deployments.
```

Comment:
One more disadvantage I see:
* sessions are destroyed when containers restart (i.e. during a release)

Reply:
That is the reason of the user journey disruption, not a second reason. perhaps the sentence can be improved to  ...
User journeys can be disrupted by sub-optimal user journey distribution across containers when container restarts, i.e. deployments.

but also worth to clarify if its for all containers or there are individual containers in charge of managing the session.